src/pkg/provisioner/provisioner.go

```diff
@@ -45,11 +47,96 @@ func setup(dockerCredentialGCR string, systemd *systemdClient) error {
 	return nil
 }
 
-func cleanup(stateDir string) error {
+func stopServices(systemd *systemdClient) error {
+	log.Println("Stopping services...")
+	if err := systemd.stop("crash-reporter.service"); err != nil {
+		return err
+	}
+	if err := systemd.stop("crash-sender.service"); err != nil {
+		return err
+	}
+	if err := systemd.stop("device_policy_manager.service"); err != nil {
+		return err
+	}
+	if err := systemd.stop("metrics-daemon.service"); err != nil {
+		return err
+	}
+	if err := systemd.stop("update-engine.service"); err != nil {
+		return err
+	}
+	log.Println("Done stopping services.")
+	return nil
+}
+
+func zeroAllFiles(dir string) error {
+	return filepath.Walk(dir, func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return fmt.Errorf("error accessing path %q: %v", path, err)
+		}
+		if info.IsDir() {
+			return nil
+		}
+		// Truncate the file
+		f, err := os.Create(path)
+		if err != nil {
+			return err
+		}
+		if err := f.Close(); err != nil {
+			return err
+		}
+		return nil
+	})
+}
+
+func cleanupDir(dir string) error {
+	fileInfos, err := ioutil.ReadDir(dir)
+	if err != nil {
+		if os.IsNotExist(err) {
+			return nil
+		} else {
+			return err
+		}
+	}
+	for _, fi := range fileInfos {
+		if err := os.RemoveAll(filepath.Join(dir, fi.Name())); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func cleanup(rootDir, stateDir string) error {
 	log.Println("Cleaning up machine state...")
 	if err := os.RemoveAll(stateDir); err != nil {
 		return err
 	}
+	if err := os.RemoveAll(filepath.Join(rootDir, "mnt", "stateful_partition", "etc")); err != nil && !os.IsNotExist(err) {
+		return err
+	}
+	if err := cleanupDir(filepath.Join(rootDir, "var", "cache")); err != nil {
+		return err
+	}
+	if err := zeroAllFiles(filepath.Join(rootDir, "var", "log")); err != nil {
+		return err
+	}
+	if err := cleanupDir(filepath.Join(rootDir, "var", "tmp")); err != nil {
```

Comment:
nit: maybe use `cleanupDirs(dirs []string)`/ for loop over list of dirs and call `cleanupDir`

Reply:
Done